### ISSUE TITLE
finetune MiniCPM-V2.6 on 8*4090 loss=0 grad_norm=Nan

### ISSUE THREAD

**Issue by zhaoyangwei123** [modelscope/ms-swift#2436]
我在4090上利用自己的数据集finetune MiniCPM-V2.6 在8285/11043的时候会出现loss=0且grad_norm=Nan的情况，我认真检查了数据并且对感到有问题的数据进行了替换发现还是这个问题，我又在8*3090上面跑了一遍发现没有任何问题，请问是对4090的适配有问题吗？
我的命令行：
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 NPROC_PER_NODE=8 swift sft --model_type minicpm-v-v2_6-chat-int4 --model_id_or_path OpenBMB/MiniCPM-V-2_6-int4 --sft_type lora --dataset /home/wzy/disk1/MiniCPM-V/finetune/relation_datasets/standard_train_no_history.jsonl --deepspeed default-zero2 --use_flash_attn True --eval_steps 1000

{"loss": 1.43502216, "acc": 0.6400209, "grad_norm": 3.48362255, "learning_rate": 1.628e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.407947, "epoch": 0.74885679, "global_step/max_steps": "8270/11043", "percentage": "74.89%", "elapsed_time": "5h 37m 49s", "remaining_time": "1h 53m 16s"}
{"loss": 1.394207, "acc": 0.63822227, "grad_norm": 4.16534615, "learning_rate": 1.623e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40806, "epoch": 0.74930955, "global_step/max_steps": "8275/11043", "percentage": "74.93%", "elapsed_time": "5h 37m 56s", "remaining_time": "1h 53m 2s"}
{"loss": 2.23302879, "acc": 0.26443233, "grad_norm": NaN, "learning_rate": 1.617e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408118, "epoch": 0.7497623, "global_step/max_steps": "8280/11043", "percentage": "74.98%", "elapsed_time": "5h 38m 5s", "remaining_time": "1h 52m 49s"}
{"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.612e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40823, "epoch": 0.75021506, "global_step/max_steps": "8285/11043", "percentage": "75.02%", "elapsed_time": "5h 38m 12s", "remaining_time": "1h 52m 35s"}
{"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.606e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40834, "epoch": 0.75066781, "global_step/max_steps": "8290/11043", "percentage": "75.07%", "elapsed_time": "5h 38m 18s", "remaining_time": "1h 52m 20s"}
{"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.601e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408448, "epoch": 0.75112057, "global_step/max_steps": "8295/11043", "percentage": "75.12%", "elapsed_time": "5h 38m 25s", "remaining_time": "1h 52m 6s"}
{"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.595e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408559, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 38m 32s", "remaining_time": "1h 51m 52s"}
{"eval_loss": NaN, "eval_acc": 0.0, "eval_runtime": 46.4195, "eval_samples_per_second": 38.432, "eval_steps_per_second": 4.804, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 39m 18s", "remaining_time": "1h 52m 8s"}

**Comment by josephzpng** [comment]
> 我在4090上利用自己的数据集finetune MiniCPM-V2.6 在8285/11043的时候会出现loss=0且grad_norm=Nan的情况，我认真检查了数据并且对感到有问题的数据进行了替换发现还是这个问题，我又在8*3090上面跑了一遍发现没有任何问题，请问是对4090的适配有问题吗？ 我的命令行： CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 NPROC_PER_NODE=8 swift sft --model_type minicpm-v-v2_6-chat-int4 --model_id_or_path OpenBMB/MiniCPM-V-2_6-int4 --sft_type lora --dataset /home/wzy/disk1/MiniCPM-V/finetune/relation_datasets/standard_train_no_history.jsonl --deepspeed default-zero2 --use_flash_attn True --eval_steps 1000
> 
> {"loss": 1.43502216, "acc": 0.6400209, "grad_norm": 3.48362255, "learning_rate": 1.628e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.407947, "epoch": 0.74885679, "global_step/max_steps": "8270/11043", "percentage": "74.89%", "elapsed_time": "5h 37m 49s", "remaining_time": "1h 53m 16s"} {"loss": 1.394207, "acc": 0.63822227, "grad_norm": 4.16534615, "learning_rate": 1.623e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40806, "epoch": 0.74930955, "global_step/max_steps": "8275/11043", "percentage": "74.93%", "elapsed_time": "5h 37m 56s", "remaining_time": "1h 53m 2s"} {"loss": 2.23302879, "acc": 0.26443233, "grad_norm": NaN, "learning_rate": 1.617e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408118, "epoch": 0.7497623, "global_step/max_steps": "8280/11043", "percentage": "74.98%", "elapsed_time": "5h 38m 5s", "remaining_time": "1h 52m 49s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.612e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40823, "epoch": 0.75021506, "global_step/max_steps": "8285/11043", "percentage": "75.02%", "elapsed_time": "5h 38m 12s", "remaining_time": "1h 52m 35s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.606e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40834, "epoch": 0.75066781, "global_step/max_steps": "8290/11043", "percentage": "75.07%", "elapsed_time": "5h 38m 18s", "remaining_time": "1h 52m 20s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.601e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408448, "epoch": 0.75112057, "global_step/max_steps": "8295/11043", "percentage": "75.12%", "elapsed_time": "5h 38m 25s", "remaining_time": "1h 52m 6s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.595e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408559, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 38m 32s", "remaining_time": "1h 51m 52s"} {"eval_loss": NaN, "eval_acc": 0.0, "eval_runtime": 46.4195, "eval_samples_per_second": 38.432, "eval_steps_per_second": 4.804, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 39m 18s", "remaining_time": "1h 52m 8s"}

Hello, have you solved it? I also encountered the same problem. When I started training, loss and grad_norm were always 0.

**Comment by zhaoyangwei123** [comment]
> > 我在4090上利用自己的数据集finetune MiniCPM-V2.6 在8285/11043的时候会出现loss=0且grad_norm=Nan的情况，我认真检查了数据并且对感到有问题的数据进行了替换发现还是这个问题，我又在8*3090上面跑了一遍发现没有任何问题，请问是对4090的适配有问题吗？ 我的命令行： CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 NPROC_PER_NODE=8 swift sft --model_type minicpm-v-v2_6-chat-int4 --model_id_or_path OpenBMB/MiniCPM-V-2_6-int4 --sft_type lora --dataset /home/wzy/disk1/MiniCPM-V/finetune/relation_datasets/standard_train_no_history.jsonl --deepspeed default-zero2 --use_flash_attn True --eval_steps 1000
> > {"loss": 1.43502216, "acc": 0.6400209, "grad_norm": 3.48362255, "learning_rate": 1.628e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.407947, "epoch": 0.74885679, "global_step/max_steps": "8270/11043", "percentage": "74.89%", "elapsed_time": "5h 37m 49s", "remaining_time": "1h 53m 16s"} {"loss": 1.394207, "acc": 0.63822227, "grad_norm": 4.16534615, "learning_rate": 1.623e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40806, "epoch": 0.74930955, "global_step/max_steps": "8275/11043", "percentage": "74.93%", "elapsed_time": "5h 37m 56s", "remaining_time": "1h 53m 2s"} {"loss": 2.23302879, "acc": 0.26443233, "grad_norm": NaN, "learning_rate": 1.617e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408118, "epoch": 0.7497623, "global_step/max_steps": "8280/11043", "percentage": "74.98%", "elapsed_time": "5h 38m 5s", "remaining_time": "1h 52m 49s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.612e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40823, "epoch": 0.75021506, "global_step/max_steps": "8285/11043", "percentage": "75.02%", "elapsed_time": "5h 38m 12s", "remaining_time": "1h 52m 35s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.606e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.40834, "epoch": 0.75066781, "global_step/max_steps": "8290/11043", "percentage": "75.07%", "elapsed_time": "5h 38m 18s", "remaining_time": "1h 52m 20s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.601e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408448, "epoch": 0.75112057, "global_step/max_steps": "8295/11043", "percentage": "75.12%", "elapsed_time": "5h 38m 25s", "remaining_time": "1h 52m 6s"} {"loss": 0.0, "acc": 0.0, "grad_norm": NaN, "learning_rate": 1.595e-05, "memory(GiB)": 19.54, "train_speed(iter/s)": 0.408559, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 38m 32s", "remaining_time": "1h 51m 52s"} {"eval_loss": NaN, "eval_acc": 0.0, "eval_runtime": 46.4195, "eval_samples_per_second": 38.432, "eval_steps_per_second": 4.804, "epoch": 0.75157332, "global_step/max_steps": "8300/11043", "percentage": "75.16%", "elapsed_time": "5h 39m 18s", "remaining_time": "1h 52m 8s"}
> 
> Hello, have you solved it? I also encountered the same problem. When I started training, loss and grad_norm were always 0.

I haven't solved this problem yet, but I think it's very likely caused by bfloat 16, because MiniCPM-V2.6 defaults to enabling bfloat on 4090, so it may be due to a numerical error to some extent. However, when I tried to change it to fp16 to finetune, I found that 4090's VRAM was not enough

**Comment by Jintao-Huang** [comment]
https://github.com/modelscope/ms-swift/issues/1175

My impression is that fp16 can lead to NaN results.

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
